### PR TITLE
HBASE-26645 Upgrade create-release to support both JDK8 and JDK11

### PR DESCRIPTION
**Pull request by ndimiduk**
update create-release Dockerfile to use jdk11-headless

Requires #3974

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 26s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  6s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   1m 49s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests |  |
| uname | Linux b82ac2d690b7 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 70cb9b0ba9 |
| Max. process+thread count | 51 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  3s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  6s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests |  |
| uname | Linux 7c5e572a8a2a 4.15.0-163-generic #171-Ubuntu SMP Fri Nov 5 11:55:11 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 70cb9b0ba9 |
| Max. process+thread count | 46 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   4m 29s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  hadolint  |   0m  1s |  There were no new hadolint issues.  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   6m  9s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests | dupname asflicense hadolint shellcheck shelldocs |
| uname | Linux 91aaddc9af87 4.15.0-161-generic #169-Ubuntu SMP Fri Oct 15 13:41:54 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 70cb9b0ba9 |
| Max. process+thread count | 46 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/console |
| versions | git=2.17.1 maven=3.6.3 shellcheck=0.4.6 hadolint=1.17.5-0-g443423c |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: dev-support/create-release/hbase-rm/Dockerfile [hunk 1]
```diff
@@ -34,13 +34,13 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get -qq -y update \
     libcurl4-openssl-dev='7.58.0-*' \
     libxml2-dev='2.9.4+dfsg1-*' \
     lsof='4.89+dfsg-*' \
-    openjdk-8-jdk='8u*' \
+    openjdk-11-jdk-headless='11*' \
```
Comment (madrob) [inline dev-support/create-release/hbase-rm/Dockerfile:37]: Update the comment at line 21?

**Review by joshelser — APPROVED** [review]

**Comment by ndimiduk** [comment]
Canceling patch as this is blocked by a bug in javac. See the issue linked from JIRA.

**Comment by ndimiduk** [comment]
@Apache9 's efforts on migrating our JDK8 dependencies out to hbase-thirdparty means that we can now use `maven.compiler.release=8`, which he also did. Bump the JDK version we use in our create-release automation. Mind taking another look?

**Comment by ndimiduk** [comment]
Or maybe we cannot do this because these tools are used for creating releases from hbase-thirdparty. Is that the case?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 42s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  3s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   1m 38s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests |  |
| uname | Linux d0f45e23ad24 5.4.0-1025-aws #25~18.04.1-Ubuntu SMP Fri Sep 11 12:03:04 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / f51e57908f |
| Max. process+thread count | 36 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  6s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  4s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m  5s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests |  |
| uname | Linux 086d5613716f 5.4.0-1025-aws #25~18.04.1-Ubuntu SMP Fri Sep 11 12:03:04 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / f51e57908f |
| Max. process+thread count | 30 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  hadolint  |   0m  2s |  There were no new hadolint issues.  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   5m 14s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests | dupname asflicense hadolint shellcheck shelldocs |
| uname | Linux 0dc32920187b 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / f51e57908f |
| Max. process+thread count | 28 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/1/console |
| versions | git=2.17.1 maven=3.6.3 shellcheck=0.4.6 hadolint=1.17.5-0-g443423c |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
> Or maybe we cannot do this because these tools are used for creating releases from hbase-thirdparty. Is that the case?

For hbase-thirdparty we still need to use JDK8. But anyway, we could install both in the docker file, and if it is for hbase we will use JDK11, for others we will use JDK8?

**Comment by ndimiduk** [comment]
Okay, this change just got complicated. Now we include both JDK versions in the docker file (the easy part) and support the user specifying which JDK they want used for the build (the annoyingly involved part).

And I pushed without doing enough testing. The new parsing code has some bugs. Will be back.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  8s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  5s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m 19s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests |  |
| uname | Linux 1c9daec0dbb7 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 41ae90d63f |
| Max. process+thread count | 41 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 37s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  3s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   1m 22s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests |  |
| uname | Linux e71a4689006e 5.4.0-1071-aws #76~18.04.1-Ubuntu SMP Mon Mar 28 17:49:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 41ae90d63f |
| Max. process+thread count | 30 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 22s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  1s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  8s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  4s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  hadolint  |   0m  0s |  There were no new hadolint issues.  |
| +1 :green_heart: |  shellcheck  |   0m  1s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   2m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests | dupname asflicense shellcheck shelldocs hadolint |
| uname | Linux 9b855182a138 5.4.0-90-generic #101-Ubuntu SMP Fri Oct 15 20:00:55 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 41ae90d63f |
| Max. process+thread count | 28 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/2/console |
| versions | git=2.17.1 maven=3.6.3 shellcheck=0.4.6 hadolint=1.17.5-0-g443423c |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
https://issues.apache.org/jira/browse/HBASE-25983

Seems we can not do release with JDK11 because we can not build the javadoc...

**Comment by ndimiduk** [comment]
> https://issues.apache.org/jira/browse/HBASE-25983
> 
> Seems we can not do release with JDK11 because we can not build the javadoc...

I saw that. This PR can sit until we have more progress, or I can close it for now, if it's in the way. As you like.

**Comment by apurtell** [comment]
We can leave it open I think, assuming HBASE-25983 is going to be fixed in the near term.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  4s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m 23s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/3/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests |  |
| uname | Linux b26e958212cd 5.4.0-1085-aws #92~18.04.1-Ubuntu SMP Wed Aug 31 17:21:08 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / b1a648ffea |
| Max. process+thread count | 33 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/3/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  4s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m 26s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests |  |
| uname | Linux 3fe7dd093963 5.4.0-1085-aws #92~18.04.1-Ubuntu SMP Wed Aug 31 17:21:08 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / b1a648ffea |
| Max. process+thread count | 37 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/3/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 20s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  spotless  |   0m 54s |  branch has no errors when running spotless:check.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  6s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  hadolint  |   0m  0s |  There were no new hadolint issues.  |
| +1 :green_heart: |  shellcheck  |   0m  3s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  spotless  |   1m  0s |  patch has no errors when running spotless:check.  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   4m 45s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests | dupname asflicense spotless shellcheck shelldocs hadolint |
| uname | Linux cf342dc61aca 5.4.0-124-generic #140-Ubuntu SMP Thu Aug 4 02:23:37 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / b1a648ffea |
| Max. process+thread count | 44 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/3/console |
| versions | git=2.34.1 maven=3.8.6 shellcheck=0.8.0 hadolint=Haskell Dockerfile Linter 2.10.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 17s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  4s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  4s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m 35s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/4/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests |  |
| uname | Linux 988e48af49a5 5.4.0-131-generic #147-Ubuntu SMP Fri Oct 14 17:07:22 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 260cd485db |
| Max. process+thread count | 38 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/4/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  4s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  1s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  spotless  |   1m  2s |  branch has no errors when running spotless:check.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  6s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  hadolint  |   0m  1s |  There were no new hadolint issues.  |
| +1 :green_heart: |  shellcheck  |   0m  3s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  spotless  |   0m 51s |  patch has no errors when running spotless:check.  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   4m 35s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3999 |
| Optional Tests | dupname asflicense spotless shellcheck shelldocs hadolint |
| uname | Linux fa405f6dd959 5.4.0-1088-aws #96~18.04.1-Ubuntu SMP Mon Oct 17 02:57:48 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 260cd485db |
| Max. process+thread count | 43 (vs. ulimit of 30000) |
| modules | C:  U:  |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-3999/4/console |
| versions | git=2.34.1 maven=3.8.6 shellcheck=0.8.0 hadolint=Haskell Dockerfile Linter 2.10.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by ndimiduk** [comment]
Should we build two docker images, one with JDK8 and the other with JDK11 installed, have the script launch each build in the appropriate image? The alternative would be to invoke `update-java-alternatives` (with `sudo`) between the two builds within the same container image.

**Comment by Apache9** [comment]
> > For the upcoming 2.5.2 release where we plan to make both hadoop2 and hadoop3 artifacts, I do not think we can use java11 since it can not be used to build artifacts for hadoop2... Of course it is possible to build hadoop3 artifacts with java11, but anyway, we can not specify java11 only from the command line...
> 
> I think you're right. Doing this with `create-release` will be strange, the way it current considers a single jvm version as "all or nothing". It's also annoying because of how ubuntu manages multiple openjdk package installs... you can't just set `JAVA_HOME` to get a java version, the `java` binary appears to be pinned to it's install location. See my latest commit on this branch, where I start working with `jenv` to enable switching JVMs more dynamically.
> 
> How do you want to proceed for 2.5.2?

Just use java8 this time. We still want to keep the java8 compatibility even for hadoop3 so this sohuld not be a problem for now.

And for multi jdks, I think we could follow what we have done in the pre commit docker image, just download Temurin(used to be called Adopt) OpenJDKs for both java8 and java11, add JAVA_HOME/bin to path, and switch JAVA_HOME if we want to use different jdks.

Maybe later, we could use java8 to build hadoop2 related artifacts, and use java11 to build hadoop3 related artifacts. And for the upcoming 3.x releases, since they only support hadoop3, I think we could make them java11 only.

Thanks.

**Pull request closed without merging**